pkgs/development/libraries/gsignond/plugins/oauth.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchFromGitLab, fetchpatch, pkgconfig, meson, ninja, glib, gsignond, check
+, json-glib, libsoup, gnutls, gtk-doc, docbook_xml_dtd_43, docbook_xml_dtd_45
+, docbook_xsl, glibcLocales, gobjectIntrospection }:
+
+stdenv.mkDerivation rec {
+  name = "gsignond-plugin-oauth-${version}";
+  version = "2018-10-04";
+
+  src = fetchFromGitLab {
+    owner = "accounts-sso";
+    repo = "gsignond-plugin-oa";
+    rev = "f80a390533587e02538b3b80be67bb88b0735a7a";
+    sha256 = "0yg7sha9ig6rliii68j2y831zzzfgg01xkkd91gzp1f8s57p5fiz";
+  };
+
+  # Fixes building documentation with meson > 0.47.0 - See -> https://gitlab.com/accounts-sso/gsignond-plugin-oa/issues/8
+  # patches = [
+  #   (fetchpatch {
+  #     url = "https://gitlab.com/worldofpeace/gsignond-plugin-oa/commit/fb121ae176468b056a0cbfebddc341d5c3c30d69.patch";
+  #     sha256 = "0hshx3sk5zzh22qkj6k87qx4xkghzk4m5qlnsz3a7iaps5dbbz7k";
+  #   })
+  # ];
+
+  nativeBuildInputs = [
+    docbook_xml_dtd_43
+    docbook_xml_dtd_45
+    docbook_xsl
+    glibcLocales
+    gobjectIntrospection
+    gtk-doc
+    meson
+    ninja
+    pkgconfig
+  ];
+
+  buildInputs = [
+    glib
+    gnutls
+    gsignond
+    json-glib
+    libsoup
+  ];
+
+  checkInputs = [ check ];
+  doCheck = true;
+
+  checkPhase = ''
```

Comment:
Why explicit declaration?

Reply:
Without it ninja just hangs.

Reply:
Weird, `meson test` should just run `ninja test` under the hood: https://github.com/mesonbuild/meson/blob/5715284177cbfdea7e1190d5b6652a2d9292ad2a/docs/markdown/Unit-tests.md#testing-tool

Reply:
I'm noticing other weirdness here when using `checkInputs`. 
So I'm going to opt to not use it to lessen complexity.